main.js

```diff
@@ -172,6 +172,12 @@ app.on('window-all-closed', function () {
   }
 })
 
+app.on('activate', function() {
+  if (!Object.keys(windows).length) {
+    init()
```

Comment:
Ah yeah. So you can have windows that aren't visible (which you would use in place of say a web worker, since those create new threads in process which aren't supported). If you close the last visible window on osx, you'd want to stay open, and close on other OSes (the background windows are not necessary anymore). But those can be left open on osx to avoid the startup time with them when reopening a visible window so it's ready faster. I didn't _see_ any non-visible windows anywhere in the code, so i just checked that all windows were closed, and that seemed to work. 

As an aside, my knowledge is probably a few months out of date depending on how much it's been changing lately, but I'd be happy to help with any api or environment questions whenever you have them 👍